.codespellrc

```diff
@@ -1,5 +1,5 @@
 [codespell]
-skip = .git,node_modules,dist,venv,.venv,.mypy_cache,.tox,yarn.lock,CHANGELOG.md,./htmlcov,*.pyc,*.log
+skip = .git,node_modules,dist,venv,.venv,.mypy_cache,.tox,yarn.lock,CHANGELOG.md,./htmlcov,*.pyc,*.log,./doc
```

Comment:
why do we want to allow typos in `doc/`? https://github.com/dandi/dandi-archive/pull/1801/commits/e132b29d74ba12d76c9db51f4568dd90334d3a87 gives no reasoning.

Reply:
I see -- it was because of the hit on the rare "therefrom" as a single word. Ok, we can either exclude it explicitly via

```patch
diff --git a/.codespellrc b/.codespellrc
index 624b90fd..edc3ab29 100644
--- a/.codespellrc
+++ b/.codespellrc
@@ -3,4 +3,4 @@ skip = .git,node_modules,dist,venv,.venv,.mypy_cache,.tox,yarn.lock,CHANGELOG.md
 # Disabled until https://github.com/codespell-project/codespell/issues/2727
 # got answer/re-solution
 # dictionary = .codespell_dict
-# ignore-words-list = 
+ignore-words-list = therefrom
```

or just decide for us to not use "rare" dictionary and only the one with clear typos:

```patch
diff --git a/.codespellrc b/.codespellrc
index 624b90fd..87d72897 100644
--- a/.codespellrc
+++ b/.codespellrc
@@ -1,6 +1,7 @@
 [codespell]
-skip = .git,node_modules,dist,venv,.venv,.mypy_cache,.tox,yarn.lock,CHANGELOG.md,./htmlcov,*.pyc,*.log,./doc
+skip = .git,node_modules,dist,venv,.venv,.mypy_cache,.tox,yarn.lock,CHANGELOG.md,./htmlcov,*.pyc,*.log
 # Disabled until https://github.com/codespell-project/codespell/issues/2727
 # got answer/re-solution
 # dictionary = .codespell_dict
+builtin = clear
 # ignore-words-list = 
```

I am ok with either with smaller preference to former as of being specific (I rarely run into those "rare" words)

Reply:
codespell, from its own README, is "designed primarily for checking misspelled words in source code". And despite the claim that "it shouldn't generate false-positives when you use a niche term it doesn't know about", it complained about the word "therefrom", which is very much a properly spelled English word.

More generally, I don't really care if typos creep into design documents--that generally won't impede the use of design docs for the reasons we want to use them, and I think preventing merge for reasons of a misspelling is much more trouble than it's worth, especially when codespell is in fact giving us false positives like it did here.

Reply:
codespell by default excludes "rare" words? Why?

If we use that `builtin = clear`, it will not flag words like "therefrom"? If that's the case, I am willing to try that, though I remain skeptical of the value of slowing down the writing of design docs because of CI-based spellcheck.